### ISSUE TITLE
Issue with some kernel modules

### ISSUE THREAD

**Issue by mchataigner** [open-telemetry/opentelemetry-ebpf-profiler#157]
Hello,
I’m trying to run the agent on my server but it crashes during startup.
The problem seems to come from one kernel module not being mapped correctly:

```
Failed to load eBPF tracer: failed to read kernel modules: addresses from modules is zero - check process permissions: 'nls_ucs2_utils 8192 1 cifs, Live 0x0000000000000000' 
```

and indeed, if I have a look at the `/proc/modules` file as root, I can see all modules having a `valid` address and this module `nls_ucs2_utils` having address `0x0000000000000000`.

My understanding is that this check is here for permission validation, could it be possible not to fail if only few modules have address 0x0 but the remaining modules having a `valid` address?

**Comment by mchataigner** [comment]
Sure, it’s a proxmox server (using Debian 12) with fairly recent kernel `6.8.12`

```sh
# cat /sys/module/nls_ucs2_utils/sections/.text
cat: /sys/module/nls_ucs2_utils/sections/.text: No such file or directory
```

```sh
# cat /proc/sys/kernel/kptr_restrict
0
```

```sh
# ls -la /sys/module/nls_ucs2_utils/sections/
total 0
drwxr-xr-x 2 root root  0 Sep 16 13:01 .
drwxr-xr-x 5 root root  0 Sep 14 14:04 ..
-r-------- 1 root root 19 Sep 16 13:01 .data
-r-------- 1 root root 19 Sep 16 13:01 .gnu.linkonce.this_module
-r-------- 1 root root 19 Sep 16 13:01 __kcrctab_gpl
-r-------- 1 root root 19 Sep 16 13:01 __ksymtab_gpl
-r-------- 1 root root 19 Sep 16 13:01 __ksymtab_strings
-r-------- 1 root root 19 Sep 16 13:01 .note.gnu.build-id
-r-------- 1 root root 19 Sep 16 13:01 .note.Linux
-r-------- 1 root root 19 Sep 16 13:01 .rodata
-r-------- 1 root root 19 Sep 16 13:01 .strtab
-r-------- 1 root root 19 Sep 16 13:01 .symtab
```

```sh
# uname -a
Linux homelab 6.8.12-1-pve #1 SMP PREEMPT_DYNAMIC PMX 6.8.12-1 (2024-08-05T16:17Z) x86_64 GNU/Linux
```

```sh
# cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 12 (bookworm)"
NAME="Debian GNU/Linux"
VERSION_ID="12"
VERSION="12 (bookworm)"
VERSION_CODENAME=bookworm
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

**Comment by fabled** [comment]
Ok, seems this is a module without any code, but data only. Perhaps update the PR to mention this corner case.

**Comment by JM-elastic** [comment]
I'm having this problem, Ubuntu 24.04 Kernel 6.8 (tried a few) and 6.9.12, pf-host-agent-8.15.3-linux-x86_64 (tried 8.15.2, 8.14.3 also)

```
➜  elastic sudo pf-host-agent-8.15.3-linux-x86_64/pf-host-agent -project-id=1 -secret-token=xxxxxxxxxxx -collection-agent=5d6a2c3e5f7a46ce86bd251502f3ccd3.profiling.us-central1.gcp.cloud.es.io:443

time="2024-11-04T00:20:01.348263207-08:00" level=info msg="Starting Elastic profiling agent 8.15.3 (revision head-9c643303, build timestamp 1728548903)"
time="2024-11-04T00:20:01.589755602-08:00" level=info msg="Interpreter tracers: perl,php,python,hotspot,ruby,v8,dotnet"
time="2024-11-04T00:20:01.589793387-08:00" level=info msg="Automatically determining environment and machine ID ..."
time="2024-11-04T00:20:15.376534804-08:00" level=info msg="Environment: hardware, machine ID: 0x1a90c832145207c3"
time="2024-11-04T00:20:15.376576151-08:00" level=info msg="Assigned ProjectID: 1 HostID: 1914249959047301059"
time="2024-11-04T00:20:15.621862573-08:00" level=info msg="Reporting via 'elastic' protocol"
time="2024-11-04T00:20:15.670770018-08:00" level=info msg="Start CPU metrics"
time="2024-11-04T00:20:15.671002174-08:00" level=info msg="Start I/O metrics"
time="2024-11-04T00:20:16.180321496-08:00" level=info msg="Found offsets: task stack 0x20, pt_regs 0x3f58, tpbase 0x2468"
time="2024-11-04T00:20:16.180430868-08:00" level=info msg="Supports generic eBPF map batch operations"
time="2024-11-04T00:20:16.180461809-08:00" level=info msg="Supports LPM trie eBPF map batch operations"
time="2024-11-04T00:20:16.184148081-08:00" level=error msg="Failed to load eBPF tracer: failed to read kernel modules: addresses from modules is zero - check process permissions: 'nls_ucs2_utils 8192 1 cifs, Live 0x0000000000000000'"
```

**Comment by rockdaboot** [comment]
@JM-elastic Please ask Elastic specific questions [here](https://discuss.elastic.co/) or contact Elastic support.

But to reply on your comment's topic: This issue has been fixed with #158, which is not part of Elastic pf-host-agent 8.15.x. The coming 8.16.x version will contain this improvement.
